src/js/components/Beacon/VariantsForm.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { Col, Row } from 'antd';
+import VariantInput from './VariantInput';
+import AssemblyIdSelect from './AssemblyIdSelect';
+
+// TODOs:
+// show which fields are required without removing the ability to leave the form blank
+
+// form state has to be one of these:
+// empty (except for autofilled assemblyID)
+// chrom, start, assemblyID, end
+// chrom, start, assemblyID, ref, alt
+
+// forgiving chromosome regex
+// accepts X, Y, etc and any one- or two-digit non-zero number
+// note that, eg, polar bears have 37 pairs of chromosomes...
+const CHROMOSOME_REGEX = /^([1-9][0-9]?|X|x|Y|y|M|m|MT|mt)$/;
+
+const NUCLEOTIDES_REGEX = /^(a|c|g|t|n|A|C|G|T|N)*$/;
+const DIGITS_REGEX = /^[0-9]+$/;
```

Comment:
constant file worthy

Reply:
I'm lukewarm on this one. It's three lines used only in this component.